### PR TITLE
introduce free scale for geo_json

### PR DESCRIPTION
**Pull request by koldunovn**
Same as #193 , but for v0.1.6. Hope I did not screw this up this time :)

### CODE REVIEW

File: folium/utilities.py [hunk 1]
```diff
@@ -305,7 +305,11 @@ def base(x):
     quants = [0, 50, 75, 85, 90]
```
Comment (ocefpaf) [inline folium/utilities.py]: @koldunovn Can you get this line down (in between 308-309), so we can easily see that the two options are doing.

File: folium/folium.py [hunk 2]
```diff
@@ -913,9 +917,21 @@ def json_style(style_cnt, line_color, line_weight, line_opacity,
...
+            if freescale==False:
+                legend = leg_templ.render({'lin_min': 0,
+                           'lin_max': int(domain[-1]*1.1),
+                           'tick_labels': tick_labels,
+                           'caption': name})
+
+
+            else:
+                legend = leg_templ.render({'lin_min':  domain[0],
+                                           'lin_max':  domain[-1],
```
Comment (ocefpaf) [inline folium/folium.py]: @koldunovn you are not padding 0.1 here.  How does that look?

File: folium/folium.py [hunk 3]
```diff
@@ -892,8 +895,9 @@ def json_style(style_cnt, line_color, line_weight, line_opacity,
 
             # D3 Color scale.
             series = data[columns[1]]
-            if threshold_scale and len(threshold_scale) > 6:
-                raise ValueError
+            if freescale == False:
```
Comment (ocefpaf) [inline folium/folium.py]: Change this to `if not freesacale` (pep8)

File: folium/folium.py (hunk 2 above)
Comment (koldunovn) [inline folium/folium.py]: It look fine - examples you can see in the original PR. As I mention there padding will put positive non zero lin_min out of scale.

File: folium/folium.py [hunk 4]
```diff
@@ -804,6 +804,9 @@ def geo_json(self, geo_path=None, geo_str=None, data_out='data.json',
             keyword argument will enable conversion to GeoJSON.
         reset: boolean, default False
             Remove all current geoJSON layers, start with new layer
+        freescale: if True use free format for the scale, where min and max values
+            are taken from the data. It also allow to plot values that do not have 0 
+            minimum and are, for example, float.
```
Comment (ocefpaf) [inline folium/folium.py]: How about: `... allow to plot values < 0 and float legend labels.`

File: folium/utilities.py (hunk 1 above)
Comment (koldunovn) [inline folium/utilities.py]: I am not sure if this is hat you want...

File: folium/folium.py (hunk 2 above)
Comment (ocefpaf) [inline folium/folium.py]: OK.  `'lin_min': int(domain[0]*1.1)` was probably a copy-and-paste mistake from my original PR).

The idea was to extend the legend a little bit.  We could be smart about it and use a sign function, expand, and re-added the sign.   But the legends look good without this, so lets leave it be.

File: folium/utilities.py (hunk 1 above)
Comment (ocefpaf) [inline folium/utilities.py]: Nope.  Just get the line `quants = [0, 50, 75, 85, 90]` on top of `return [base(np.percentile(arr, x)) for x in quants]`.

Comment (koldunovn) [inline folium/utilities.py]: Oh, yes, make sence.

File: folium/folium.py (hunk 2 above)
Comment (koldunovn) [inline folium/folium.py]: Without padding it looks more "normal" to me, but it's a mater of taste.

**Comment by ocefpaf** [comment]
@koldunovn can you squash the commits?

@BibMartin and @themiurgo I am OK with this PR as it fixes the zero based legend and expand the legend use for other cases without breaking people codes. I think is a nice addition until we have `v0.2.0`

I will be grateful if any of you two can take a second look at this PR. (I am planing to merge it tomorrow morning.)

**Pull request merged**